src/Utility/InputValidator.php

```diff
@@ -0,0 +1,111 @@
+<?php
+/**
+ * Input validation utilities.
+ *
+ * @package Requests
+ * @subpackage Utilities
+ */
+
+namespace WpOrg\Requests\Utility;
+
+use ArrayAccess;
+use CurlHandle;
+use Traversable;
+
+/**
+ * Input validation utilities.
+ *
+ * @package Requests
+ * @subpackage Utilities
+ */
+final class InputValidator {
+
+	/**
+	 * Verify that a received input parameter is of type string or is "stringable".
+	 *
+	 * @param mixed $input Input parameter to verify.
+	 *
+	 * @return bool
+	 */
+	public static function is_string_or_stringable($input) {
```

Comment:
Potential alternative name:
- `is_stringable()`

Reply:
That wouldn't be correct as the method strictly checks for a string or Stringable object, while all scalars realistically are _stringable_ (will juggle without errors).